webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,279 @@
+            this.init();
+        } else if (this.props.connected && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate(): void {
+        if (this.props.isLoaded && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props): void{
+        this.setState({
+            isLoaded: true,
+            ticketDetails: data.ticketDetails,
+        });
+    }
+
+    fixVersionLabel(fixVersion: string) {
+        if (fixVersion) {
+            const fixVersionString = 'Fix Version :';
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        if (ticketStatus.toLowerCase() === 'indeterminate') {
+            return (<span className='ticket-status-indeterminate default-style'>{ticketStatus}</span>);
+        }
+
+        if (ticketStatus.toLowerCase() === 'done') {
+            return (<span className='ticket-status-done default-style'>{ticketStatus}</span>);
+        }
+
+        return (<span className='ticket-status-default default-style'>{ticketStatus}</span>);
+    }
+
+    labelList(labels: string[]) {
+        if (labels.length) {
+            let totalString = 0;
+            let totalHide = 0;
+            const labelList = labels.map((label: string, key: number): JSX.Element => {
+                if (totalString < 3) {
+                    totalString++;
+                    return (
+                        <span
+                            key={key}
+                            className='jiraticket-popover-label-list'
+                        >
+                            {label}
+                        </span>);
+                }
+                totalHide++;
+                if (key === labels.length - 1) {
+                    const moreLabels = `+${totalHide}more`;
+                    return (
+                        <span
+                            key={key}
+                            className='jiraticket-popover-label-list'
+                        >
+                            {moreLabels}
+                        </span>);
+                }
+                return <></>;
+            });
+
+            return (<div className={'ticket-popover-label'}>{labelList}</div>);
+        }
+        return null;
+    }
+
+    render() {
+        if (!this.state.isLoaded) {
+            return (<p/>);
+        }
+        const ticketDetails = this.state.ticketDetails;
+        const jiraTicketURI = this.state.href;
```

Comment:
```suggestion
        const {ticketDetails, href as jiraTicketURI} = this.state;
```

Reply:
In this case it's condsidering `jiraTicketURI` as a type. used `const {ticketDetails, href: jiraTicketURI} = this.state;`